src/containers/citationsBoxContainer.tsx

```diff
@@ -165,10 +166,11 @@ function CitationsBoxContainer(props: {
 		};
 	}, []);
 
+	// Sync state with Zotero overlay
 	useEffect(() => {
-		Zotero[config.addonInstance].data.zoteroOverlay.setSourceItem(
-			sourceItem,
-		);
+		(
+			Zotero[config.addonInstance].data.zoteroOverlay as ZoteroOverlay
```

Comment:
The fact that we have to sync things up with the Zotero overlay seems like a bit of a weird / not desirable design pattern to be honest?

Not a big deal if it works though. What do you think?

Reply:
I think this was here even before I did the other PR, so you might know better. But from what I did with the PID stuff, because we use React for separate components that are not within a unique React hierarchy, we do have to do some manual syncing. Here, because we use `this._sourceItem` for certain functions in ZoteroOverlay, we need to make sure that the object is actually in sync, otherwise it would simply act upon a copy it made when the item was selected.

Reply:
Yeah, it was already there don't worry. This just made me think about it.

But that makes sense, that they're all isolated react components so if they need to interact they need to be synchronised manually.

All good, thanks!